fix: paper updates (#351)
scikit-hep/pylhe: paper/paper.md

While interfaces for C/C++/Fortran exist in the respective generators, a lightweight and easy-to-use Python interface was missing until the inception of `pylhe` in 2015.
Additionally, `pylhe` can serve as a crucial interface for emerging machine learning applications in particle physics, allowing researchers to efficiently extract event data for training neural networks and other machine learning models used in event classification, anomaly detection, and physics analysis.

# State of the field

Unlike the existing C/C++/Fortran interfaces provided by the Monte Carlo event generators, `pylhe` offers a pure Python interface that is more accessible and easier to use.
When `pylhe` was first developed, there were no other Python libraries available for reading and writing LHE files.

> Nowadays, there are a few other smaller Python libraries with less adoption than `pylhe` that provide only read functionality, such as `lhereader`.

Reviewer: We should add a reference for lhereader, at least https://github.com/diptaparna/lhereader. This being said, we also need to state that the library is non-active since 9 years, hence effectively obsolete. (There is also the fact that I see it depend on ROOT and pyroot, so really not the right solution.)

> At this point it should also be mentioned that several LHE libraries exist in other programming languages, such as Go (`go-hep` [@Binet2017]), Rust (`lhe`, `lhef`, `event_file_reader`), Julia (`LHEF.jl`) and Haskell (`lhe.hs`) with varying degree of completeness.

Reviewer: ```suggestion
For completeness, it should be mentioned that several LHE libraries exist in other programming languages, such as Go (`go-hep` [@Binet2017]), Rust (`lhe`, `lhef`, `event_file_reader`), Julia (`LHEF.jl`) and Haskell (`lhe.hs`). These provide varying degrees of completeness.
```

# Software design

`pylhe` allows for easy reading and writing of `.lhe` and `.lhe.gz` files in Python, enabling seamless integration into modern data analysis workflows in high-energy physics.
The pythonic event yielding approach allows for memory-efficient processing of arbitrarily large LHE files by streaming events one at a time rather than loading all of them at once into memory.
Internally, `pylhe` uses `xml.etree.ElementTree` to parse the XML structure, since using the `lxml` library did not provide a significant speed up.
The library facilitates quick validation of event files through programmatic access to event structure and particle properties, making it straightforward to perform sanity checks on generated events.
This can be done for example via the integration with Awkward Array [@Pivarski_Awkward_Array_2018] through the `to_awkward()` function, which converts LHE events into columnar data structures optimized for vectorized operations and efficient analysis of large datasets.

# Research impact statement

> `pylhe` has already been used in various research projects and publications within high-energy physics.

Reviewer: ```suggestion
`pylhe` is regularly used in various research projects and publications within high-energy physics.
```
